src/objectid.ts

```diff
@@ -143,6 +143,15 @@ export class ObjectId extends BSONValue {
     }
   }
 
+  /**
+   * @internal
+   * Validates the input string is a valid hex representation of an ObjectId.
+   */
+  private static validateHexString(input: string): boolean {
+    if (input == null || input.length !== 24) return false;
+    return checkForHexRegExp.test(input);
```

Comment:
```js
function validateHexString(string) {
  if (string?.length !== 24) return false;
  for (let i = 0; i < 24; i++) {
    const char = string[i];
    if (char !== "0" || char !== "1" || char !== "2" || char !== "3" || char !== "4" || char !== "5" || char !== "6" || char !== "7" || char !== "8" || char !== "9" || char !== "a" || char !== "b" || char !== "c" || char !== "d" || char !== "e" || char !== "f" || char !== "A" || char !== "B" || char !== "C" || char !== "D" || char !== "E" || char !== "F") {
      return false;
    }
  }
  return true;
}
```

It may seem verbose but what do you think of writing out the hex validation? it appears to me that it beats out a regex check by a substantial amount, so seemingly worth the verbosity. I am curious if you get the same results?

Reply:
Does it beat the regex version if you check a string such that you always have a lot of redundant checks, like `FFFFFFF...`?

Reply:
@nbbeeken  this is a great idea. Regex really is overkill for such a constrained character set. I agree this would be worth the increased verbosity since it would also speed up deserialization and ObjectId creation.

I wonder if checking the ascii code range for each character would give us similar performance while being slightly less verbose.

Something like this maybe:

```ts
const char = string.charCodeAt(i)
// ascii 0 = 48, 9 = 57, a = 97, f = 102
If ((char >= 48 && char <= 57) || (char >= 97 && char <= 102)) {
  continue
}
return false
```

I’ll take a deeper look at this tomorrow and try out those new benchmarks 💪

Reply:
@nbbeeken I've pushed a new commit using the ascii code range checks, which is much faster than regex for invalid strings, but matches the performance for valid strings. isValid is now up to **115,000%** faster for invalid strings 😂 

I tried your suggestion above comparing each character against each possible literal but saw some regressions. 

Thanks for the suggestion!

#### Regex
| Test                        | Ops/s (Main)         | Ops/s (New)          | % Improvement   |
|-----------------------------|----------------------|----------------------|-----------------|
| bson#new ObjectId(string)    | 3,377,558 ops/sec    | 3,473,650 ops/sec    | 2.85%           |
| bson#isValid                | 5,652,304 ops/sec    | 21,290,097 ops/sec   | 276.66%         |
| bson#isValid uppercase      | 5,627,394 ops/sec    | 21,069,665 ops/sec   | 274.41%         |
| bson#isValid invalid end    | 132,198 ops/sec      | 11,397,180 ops/sec   | 8,521.29%       |
| bson#isValid invalid start  | 134,177 ops/sec      | 73,224,586 ops/sec   | 54,473.33%      |
| bson#isValid invalid FFF    | 5,841,548 ops/sec    | 22,004,532 ops/sec   | 276.69%         |

#### Literal comparison
| Test                        | Ops/s (Main)         | Ops/s (New)          | % Improvement   |
|-----------------------------|----------------------|----------------------|-----------------|
| bson#new ObjectId(string)    | 3,505,060 ops/sec    | 3,087,243 ops/sec    | -11.92%         |
| bson#isValid                | 5,581,549 ops/sec    | 12,757,135 ops/sec   | 128.56%         |
| bson#isValid uppercase      | 5,632,543 ops/sec    | 12,840,180 ops/sec   | 127.96%         |
| bson#isValid invalid end       | 132,877 ops/sec      | 13,463,762 ops/sec   | 10,032.48%      |
| bson#isValid invalid start  | 133,114 ops/sec      | 151,365,677 ops/sec  | 113,611.27%     |
| bson#isValid invalid FFF    | 5,889,353 ops/sec    | 8,851,861 ops/sec    | 50.30%          |

#### ASCII Range
| Test                        | Ops/s (Main)         | Ops/s (New)          | % Improvement   |
|-----------------------------|----------------------|----------------------|-----------------|
| bson#new ObjectId(string)    | 3,173,643 ops/sec    | 3,299,153 ops/sec    | 3.95%           |
| bson#isValid                | 5,513,041 ops/sec    | 22,529,639 ops/sec   | 308.66%         |
| bson#isValid uppercase      | 5,498,098 ops/sec    | 21,776,208 ops/sec   | 296.07%         |
| bson#isValid invalid end       | 131,657 ops/sec      | 22,592,719 ops/sec   | 17,060.24%      |
| bson#isValid invalid start  | 132,751 ops/sec      | 152,311,298 ops/sec  | 114,634.47%     |
| bson#isValid invalid FFF    | 5,800,881 ops/sec    | 20,653,640 ops/sec   | 256.04%         |


- cpu: Apple M1 Pro
- cores: 10
- arch: arm64
- os: darwin (23.6.0)
- ram: 32GB
- node: 20.17.0

> Suggestion: You might want to also output the node version that is being used to run the benchmarks. I see big differences b/w 18,20,22